src/modules/form/components/DynamicFormFields.tsx

```diff
@@ -29,7 +29,7 @@ const DynamicFormFields: React.FC<Props> = ({ handlers, ...props }) => {
       let value = baseValue;
       // Convert string values to appropriate number types
       if (item) {
-        if (item.type === ItemType.Integer) value = parseInt(value) || 0;
+        if (item.type === ItemType.Integer) value = parseInt(value) || null;
```

Comment:
@gigxz let's definitely discuss this change before merging. I am a bit hesitant about what existing behavior might break with this change, but in this specific case, it seems like we would not want to save a 0 when the fetched value is null and/or unparseable. (0 being an allowable value for this score.)

Reply:
Yes lets extract this change, it feels risky. I don't disagree-- forcing it to zero seems odd-- but I don't feel comfortable with making such a broad change as part of a very specific custom integration work. Can you find a workaround for AHA? Maybe we can save as -1 to indicate that there was no score?

Reply:
Sure, I've reverted it. That workaround seems reasonable. This is in the `itemChanged` callback, so it's not called anymore unless a score is returned. https://github.com/greenriver/hmis-frontend/pull/1202/commits/ab5985993761937478ad94b522bff28aaa531cc3